enhancements/proxy-passdown.md

```diff
@@ -0,0 +1,162 @@
+   2. Operator SDK features and documents to facilitate implementation of
+      proxy-friendly operators.
+
+### Non-Goals
+
+   1. It would be nice to encourage proxy friendly operators by
+      including that functionality as a badge or part of the maturity
+      model on operatorhub, but that is out of scope for this proposal.
+   2. Automatic scaffolding of proxy passdown cannot be done because the
+      logic must occur in operator-specific code.
+   3. It is beyond the scope of this proposal to set up trusted certs
+      required for https connections to a proxy.
+
+
+### How the Operator receives Env Vars
+
+In all cases, proxy environment variables to be passed to operands must
+be included on the Operator deployment. This can be handled by the
+administrator, or automatically set by OLM.
+
+### How the Operator passes Env vars to Operands
+
+For Go-based operators, set of functions will be created in operator-lib
+(and offered upstream to controller-runtime) to read the proxy variables
+from the Operator environment and inject these variables into each of
+the container specs of various kubernetes objects. Operator authors will
+be responsible for invoking these functions. We will need functions for:
+
+   0. operatorlib.ReadProxyVarsFromEnv() // Helper function
+   1. operatorlib.AddProxyToDeployment(dep *appsv1.Deployment)
+   2. operatorlib.AddProxyToDaemonSets(daem *appsv1.DaemonSet)
+   3. operatorlib.AddProxyToReplicaSets(rep *appsv1.ReplicaSet)
+   4. operatorlib.AddProxyToPods(pod *appsv1.Pod)
+
+Simplified Code:
+
+```go
+// Read environment variables for "HTTP_PROXY, HTTPS_PROXY, and
+// NO_PROXY"
+func ReadProxyVarsFromEnv() ([]corev1.EnvVar) {
+   httpProxyEnvVar := "HTTP_PROXY"
+   httpproxy := os.Getenv("HTTP_PROXY")
+   return []corev1.EnvVar{
+       {
+           Name:  httpProxyEnvVar,
+           Value: httpproxy,
+       },
+       {
+           Name:  strings.ToLower(httpProxyEnvVar),
+           Value: httpproxy,
+       },
+    }
+}
+
+func AddProxyToDeployment(dep *appsv1.Deployment) {
+    // Create []corev1.EnvVar from Operator environment
+    proxyVars := ReadProxyVarsFromEnv()
+    // Pass proxy vars to each container
+   for _, container := range dep.Spec.Template.Spec.Containers {
+       container.Env = append(container.Env, proxyVars)
+   }
+}
+```
+
+
+Example usage with the [memcached operator](https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/controllers/memcached_controller.go#L83)
+
+```go
+dep := r.deploymentForMemcached(memcached)
+operatorlib.AddProxyToDeployment(dep)
+```
+
+
+For Ansible-based operators, the simplest way is to lookup and pass the
+environment variables directly where the objects are created by Operator
+authors. For example, the Ansible memcached `deployment.yaml` would
+include:
+```yaml
+   env:
+      HTTPS_PROXY: {{ lookup('env', 'HTTPS_PROXY') | default('', True) }}'
```

Comment:
TODO: make it clear that this env is in the container spec of deployments/pods/etc and NOT a part of the Ansible env

Reply:
I don't think it will provide a good use experience of ansible/helm operator for customers. The auther can only add one cluster proxy. And the bundle image can't be reused for other clusters. Why not use the way to add the cluster proxy of OLM. We shouldn't add the proxy config during the "build" operators. Should add the proxy config during the "run" operators.